src/utils/middleware.ts

```diff
@@ -259,7 +259,7 @@ export function makeSpidStrategy(
   doneCb?: DoneCallbackT
 ): SpidStrategy {
   return new SpidStrategy(
-    { ...options, passReqToCallback: true },
+    { ...options.sp, passReqToCallback: true },
```

Comment:
It seems to me we were passing wrong parameters since always, as this library is currently used in production. Isn't it?

So my questions are:
* how does it work now?
* are we sure we're not gonna break working behaviours?

Reply:
Yes, this library is currently in production and this error was present from the beginning.
In production works because the `SpidStrategy` use a `CustomSAMLClient` that override the native `SamlClient` of the `saml-strategy` at every `authenticare` request
```typescript
const samlService = new CustomSamlClient(
  {
    ...this.options, // ------> The wrong options
    ...samlOptions // ------> The options loaded with getSamlOptions
  },
  this.extendedRedisCacheProvider,
  this.tamperAuthorizeRequest,
  this.preValidateResponse,
  (...args) => (this.doneCb ? this.doneCb(req.ip, ...args) : undefined)
);
const strategy = Object.setPrototypeOf(
  {
    ...this,
    _saml: samlService // ------> Override the native saml client
  },
  this
);
super.authenticate.call(strategy, req, options);
```
[[link](https://github.com/pagopa/io-spid-commons/blob/IC-241-spidstrategy-constructor-parameter-fix/src/strategy/spid.ts#L78)]

The native client is created from the wrong `SamlOption`, the only properties affected from the bug seems to be the name property of `SamlStrategy`, because the native `_saml` is overridden
```javascript
// Customizing the name can be useful to support multiple SAML configurations at the same time.
// Unlike other options, this one gets deleted instead of passed along.
if  (options.name) {
  this.name  = options.name;
}
else {
  this.name = 'saml';
}
```
[[strategy.js in passport-saml](https://github.com/node-saml/passport-saml/blob/v1.3.5/lib/passport-saml/strategy.js#L6)]